grpc/src/main/java/com/avast/metrics/grpc/GrpcServerMonitoringInterceptor.java

```diff
@@ -0,0 +1,72 @@
+package com.avast.metrics.grpc;
+
+import com.avast.metrics.api.Monitor;
+import io.grpc.*;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+
+public class GrpcServerMonitoringInterceptor implements ServerInterceptor {
+    private final Monitor monitor;
+    private final Clock clock;
+
+    public GrpcServerMonitoringInterceptor(final Monitor monitor, final Clock clock) {
+        this.monitor = monitor;
+        this.clock = clock;
+    }
+
+    protected GrpcServerMonitoringInterceptor(final Monitor monitor) {
+        this.monitor = monitor;
+        this.clock = Clock.systemUTC();
+    }
+
+    @Override
+    public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(final ServerCall<ReqT, RespT> call, final Metadata headers, final ServerCallHandler<ReqT, RespT> next) {
+        final Instant start = clock.instant();
+
+        final ServerCall<ReqT, RespT> newCall = new ServerCall<ReqT, RespT>() {
```

Comment:
The is IMHO something like `ForwardingServerCall.SimpleForwardingServerCall` - it would simplify the code.

Reply:
OK